### ISSUE TITLE
Able to get extended acl?

### ISSUE THREAD

**Issue by lordlala** [Real-Gecko/Filemin#74]
Hi. On the old java file manager you could manage the extended acl's or whatever. Is this possible or going to be possible with filemin?

**Comment by Real-Gecko** [comment]
I did not check exact Java FM ACL.
Can you describe what exactly you want to get?

**Comment by lordlala** [comment]
Sorry for the delay.

In the java file browser, there is a button labeled "ACL" which allows you to manipulate the extended ACL properties of a file or folder. See attached image.
![aclexample](https://cloud.githubusercontent.com/assets/14075692/10823307/777cd18a-7e31-11e5-9587-d3c696d30051.png)

**Comment by lordlala** [comment]
Also, maybe I am doing something wrong or it just isn't designed to do this, but when I select a folder and click on the settings icon (chmod selected), it does not pull in the current permissions. Same with the chown button. Is this by design? I would like to be able to see what the current permission are, before making changes, so it would be helpful if it pre-filled the window with the current info. I hope that makes sense.

**Comment by Real-Gecko** [comment]
I don't see any unusal on screenshot that's not implemented in current version of Filemin. Maybe it is something in dropdown menu?
And yes, ops like "chmod" or "chown" are multientry, that means you can select ten folders and twenty files and chmod/chown them in one go. However in such a scenario it is impossible to get current permissions/owner cause it may vary across all selected items. This issue will be resolved after I implement context menu, allowing singular entry actions.

**Comment by Real-Gecko** [comment]
Closing due to no activity.

**Comment by dv-anomaly** [comment]
ACL lets you set permissions on a per user / group basis not just the owner/group/other of a directory or file.  https://help.ubuntu.com/community/FilePermissionsACLs Filemin does not seem to have this functionality?

For example I have a directory on my NAS with the following ACLs:
```
dh4@dh4-nas:~$ getfacl /pool/backups
getfacl: Removing leading '/' from absolute path names
# file: pool/backups
# owner: root
# group: root
user::rwx
user:dh4:rwx
group::---
mask::rwx
other::---

```
notice the user dh4 has their own permissions while root is the owner and only user with any access to the directory. ACL allows dh4 to also have read/write access. It would be nice to have a graphical way to add, remove, and set permissions for a particular user or group using posixacl.

**Comment by Real-Gecko** [comment]
Latest version of Filemin supports SELinux contexts.
I never worked with ACL so adding this may take some time.

**Comment by dv-anomaly** [comment]
No worries, thank you for considering the feature. The example above is pretty basic, but as you can imagine provides endless possibilities for permissions.

**Comment by Real-Gecko** [comment]
Yeah, seems cool, especially when you deal with shared hosting :)

**Comment by dv-anomaly** [comment]
I'd like to put a $25 USD bounty on this to any developer willing to get this implemented.  My requirement is 100% compatibility with posixacl, the ability to set permissions for individual users and groups.

**Comment by iliaross** [comment]
It's a cool feature to have. I like it.

Old Java File Manager had full implementation if it?

**Comment by dv-anomaly** [comment]
You can take a look at this existing module. 
 http://webmin-fsacls.sourceforge.net/en/

The interface leaves much to be desired, but it works. This should be integrated into the permissions management window or add a new context menu for acls when right-clicking an item.

**Comment by iliaross** [comment]
I will, thanks.

**Comment by dv-anomaly** [comment]
I'll give it until the end of the month if you or another developer would prefer to receive the donation directly with Paypal. Otherwise I'll through it up on bountysource at the end of the month, which means they'll take that 10% fee. Thanks everyone, just let me know if you have any questions.

@qooob Yes I think the java file manager did have full compatibility as well. I haven't used it but based on screenshots it would appear so.

**Comment by iliaross** [comment]
I'm just interested in doing it..

..not doing it for $25 😂

Before the end of the month? Maybe. Can't promise though.. There is a lot on my todo.

**Comment by dv-anomaly** [comment]
Well if you can always donate it back into the project or another project of you want. Either way defiantly claim it when your done. It's my pleasure to help out developers / projects when I can.

**Comment by Real-Gecko** [comment]
Closing due to inactivity, feel free to repoen if still actual.

**Comment by dhmetros** [comment]
This issue seems to have dropped off due to lack of activity.  This is a really nice feature to have.  We would greatly benefit from it.  OpenMediaVault has one that might be useful for design/implementation inspiration.
(Some screenshots from Google Search)
An ACL button above the file-folder list to open the ACL view/edit window:

![folder-list-acl](https://user-images.githubusercontent.com/11427602/36220533-912c4932-1180-11e8-9635-6ec264fce3cf.jpg)

The ACL window itself:
![image](https://user-images.githubusercontent.com/11427602/36220567-a9648a14-1180-11e8-96f4-0b4392337b8d.png)

It's missing some things.  For example, the file-folder list (first image) doesn't show what kind of permissions things have.  "ls", for example, would give something more like what filemin already has, plus a "+" sign showing that it has ACLs, too.  Also, listing all users in the ACL window makes it harder to parse what's there and what's not for long lists of users.  If you wanted to keep the list of users/groups, separating them out with who has ACLs applied to them at the top would help.

**Comment by RKDN** [comment]
I would love to see this implemented.

There are two commands:

getfacl - https://linux.die.net/man/1/getfacl
setfacl - https://linux.die.net/man/1/setfacl

**Comment by dv-anomaly** [comment]
And when setting permissions it should either set the default, or have an option to also set the default. The default ALC is what's inherited by child directories.

EG: `setfacl -Rm d:u:username:rwx /pool`
